### PR TITLE
Add voters count support

### PR DESCRIPTION
**Pull request by ClearlyClaire**
For newly-created polls, count voters in addition to the number of votes.
Present them in the REST API as `voters_count` and over ActivityPub as `toot:votersCount`.

When that information is available, the text now says “X voters” instead of “X votes”, and the tallies use that for percentages instead of the total number of votes (which is far less useful for `anyOf` polls) 

Marked as a draft because it probably could use more tests, and I'm very unsure about the attribute it should use.

image

### CODE REVIEW

File: app/javascript/mastodon/components/poll.js [hunk 1]
```diff
@@ -101,7 +101,8 @@ class Poll extends ImmutablePureComponent {
 
   renderOption (option, optionIndex, showResults) {
     const { poll, disabled } = this.props;
-    const percent            = poll.get('votes_count') === 0 ? 0 : (option.get('votes_count') / poll.get('votes_count')) * 100;
+    const pollVotesCount         = poll.get('voters_count') || poll.get('votes_count');
```
Comment (nightpool) [inline app/javascript/mastodon/components/poll.js]: alignment.

**Comment by Gargron** [comment]
I'm hesitant to add another custom property. Is the problem being solved the confusion around submitting another multiple-choice vote -> number of votes going up? That doesn't seem wrong to me per se.

**Comment by ThibG** [comment]
It's not just that. 40 people voting just one option each is indistinguishable from 10 people all voting all the options, although those have very different meanings ultimately

**Comment by trwnh** [comment]
instead of `votersCount` i would propose something like using `totalItems` of the `replies` Collection. whether the `Question` is `oneOf` or `anyOf` should be irrelevant; each reply should contain an array of the selected answers.

**Comment by ThibG** [comment]
@trwnh no, that would mean listing all the votes, and each vote is for one option only, so that would just be equivalent to what we currently do

**Comment by trwnh** [comment]
Wait, so one vote creates multiple responses? That doesn't make sense...

**Comment by ThibG** [comment]
Yes, a vote is a reply object with the same name as the option it is for, voting multiple options require multiple such objects.

**Comment by ThibG** [comment]
Marked as ready for review as there seem to be an agreement that the AS vocabulary isn't enough for this and pleroma devs are ok with the chosen attribute name.

**Review by Gargron — APPROVED** [review]

**Pull request merged**